snap/info.go

```diff
@@ -1075,3 +1075,13 @@ func InstanceName(snapName, instanceKey string) string {
 	}
 	return snapName
 }
+
+// ByType sorts the given slice of snap info by types. The most
+// important types will come first.
+type ByType []*Info
+
+func (r ByType) Len() int      { return len(r) }
+func (r ByType) Swap(i, j int) { r[i], r[j] = r[j], r[i] }
+func (r ByType) Less(i, j int) bool {
+	return typeOrder[r[i].Type] < typeOrder[r[j].Type]
```

Comment:
If you would be interested in taking my advice below this then becomes `r[i].Type.sortOrder() , r[j].Type.sortOrder()`